src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> node = new Node<>(null,value,null);
+            head = node;
+            tail = node;
+            size++;
+        } else {
+            Node<T> node = new Node<>(tail,value,null);
+            tail.next = node;
+            tail = node;
+            size++;
+        }
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. You've done this correctly.